test/general/test_serialization.cpp

```diff
@@ -417,12 +417,12 @@ TEST(Storage, groups)
     // create group implicitly if permissive flag is set
     EXPECT_FALSE(file->hasGroup("one"));
     EXPECT_THROW(file->checkGroup("one"), CanteraError);
-    EXPECT_TRUE(file->checkGroup("one", true));
+    EXPECT_FALSE(file->checkGroup("one", true));
     EXPECT_TRUE(file->hasGroup("one"));
```

Comment:
I think I'm missing something here -- if `checkGroup("one", [false])` raises an exception, then it can't create the group with the `permissive=true` flag either, right? But how does `hasGroup("one")` return `true` afterwards?

Reply:
The docstring was somewhat ambiguous / implementation was not clear, as the method captures both read and write modes. The docstring now reads:
```C++
    //! Check whether path location exists.
    //! If the file is accessed in write mode, the method attempts to create a new
    //! location; an exceptions is thrown if the location does not exist unless the
    //! *permissive* flag is set.
    //! @param id  storage location within file
    //! @param permissive  if true, do not raise exceptions (default=false)
    //! @returns  boolean indicating whether id is pre-existing
    bool checkGroup(const string& id, bool permissive=false);
```
Based on this,  `false` can be returned while creating a group as long as `permissive` is `true`.